$OpenBSD$
1. Un-hardcode system paths.
2. Respect OpenBSD shared library naming rules.
3. Use common GCC options to avoid duplication.
4. Fix --no-undefined usage and sync other things with current OpenBSD style.
--- qtbase/mkspecs/openbsd-g++/qmake.conf.ports.orig	Thu Sep 11 14:48:07 2014
+++ qtbase/mkspecs/openbsd-g++/qmake.conf	Thu Nov 20 22:36:15 2014
@@ -7,6 +7,9 @@ QMAKE_PLATFORM          = openbsd bsd
 
 QMAKE_COMPILER          = gcc
 
+include(../common/gcc-base-unix.conf)
+include(../common/unix.conf)
+
 QMAKE_CC                = gcc
 QMAKE_LEX               = flex
 QMAKE_LEXFLAGS          =
@@ -22,6 +25,8 @@ QMAKE_CFLAGS_SHLIB      = -fPIC
 QMAKE_CFLAGS_STATIC_LIB = $$QMAKE_CFLAGS_SHLIB
 QMAKE_CFLAGS_YACC       = -Wno-unused -Wno-parentheses
 QMAKE_CFLAGS_THREAD     = -pthread
+QMAKE_CFLAGS_OPTIMIZE_FULL = $$QMAKE_CFLAGS_RELEASE
+QMAKE_CFLAGS_RELEASE_WITH_DEBUGINFO	= $$QMAKE_CFLAGS_RELEASE -g
 
 QMAKE_CXX               = g++
 QMAKE_CXXFLAGS          = $$QMAKE_CFLAGS
@@ -34,13 +39,15 @@ QMAKE_CXXFLAGS_SHLIB    = $$QMAKE_CFLAGS_SHLIB
 QMAKE_CXXFLAGS_STATIC_LIB = $$QMAKE_CFLAGS_STATIC_LIB
 QMAKE_CXXFLAGS_YACC     = $$QMAKE_CFLAGS_YACC
 QMAKE_CXXFLAGS_THREAD   = $$QMAKE_CFLAGS_THREAD
+QMAKE_CXXFLAGS_OPTIMIZE_FULL = $$QMAKE_CXXFLAGS_RELEASE
+QMAKE_CXXFLAGS_RELEASE_WITH_DEBUGINFO	= $$QMAKE_CFLAGS_RELEASE_WITH_DEBUGINFO
 
-QMAKE_INCDIR            = /usr/local/include
-QMAKE_LIBDIR            = /usr/local/lib
-QMAKE_INCDIR_X11        = /usr/X11R6/include
-QMAKE_LIBDIR_X11        = /usr/X11R6/lib
-QMAKE_INCDIR_OPENGL     = /usr/X11R6/include
-QMAKE_LIBDIR_OPENGL     = /usr/X11R6/lib
+QMAKE_INCDIR            = $(LOCALBASE)/include
+QMAKE_LIBDIR            = $(LOCALBASE)/lib
+QMAKE_INCDIR_X11        = $(X11BASE)/include
+QMAKE_LIBDIR_X11        = $(X11BASE)/lib
+QMAKE_INCDIR_OPENGL     = $(X11BASE)/include
+QMAKE_LIBDIR_OPENGL     = $(X11BASE)/lib
 
 QMAKE_LINK              = g++
 QMAKE_LINK_SHLIB        = g++
@@ -52,22 +59,24 @@ QMAKE_LINK_SHLIB_CMD    = $$QMAKE_LINK_SHLIB $(LFLAGS)
 QMAKE_LFLAGS            =
 QMAKE_LFLAGS_RELEASE    =
 QMAKE_LFLAGS_DEBUG      =
-QMAKE_LFLAGS_SHLIB      = -shared
-QMAKE_LFLAGS_PLUGIN     = $$QMAKE_LFLAGS_SHLIB
-QMAKE_LFLAGS_SONAME     = -Wl,-soname,
-QMAKE_LFLAGS_THREAD     = -pthread
-QMAKE_LFLAGS_NOUNDEF    = -Wl,-no_unresolved
+#QMAKE_LFLAGS_SHLIB      = -shared
+#QMAKE_LFLAGS_PLUGIN     = $$QMAKE_LFLAGS_SHLIB
+#QMAKE_LFLAGS_SONAME     = -Wl,-soname,
+#QMAKE_LFLAGS_THREAD     = -pthread
+#QMAKE_LFLAGS_NOUNDEF    = -Wl,--no-undefined
+# -Wl,--no-allow-shlib-undefined
 
+QMAKE_OPENBSD_SHLIB	= 1
+
 QMAKE_LIBS              =
 QMAKE_LIBS_DYNLOAD      =
 QMAKE_LIBS_X11          = -lXext -lX11 -lm
 QMAKE_LIBS_OPENGL       = -lGL
-QMAKE_LIBS_THREAD       =
+QMAKE_LIBS_THREAD       = -lpthread
 
 QMAKE_AR                = ar q
 QMAKE_OBJCOPY           = objcopy
 QMAKE_NM                = nm -P
 QMAKE_RANLIB            = ranlib
 
-include(../common/unix.conf)
 load(qt_config)
